### ISSUE TITLE
Requirements class

### ISSUE THREAD

**Issue by digitall-it** [lerni/klaro-cookie-consent#3]
Does the integration interact in any way with the Requirements class?

`Requirements::javascript("site/javascript/data-collector.js");`

what about instead this use case?

`<% require javascript("<my-module-dir>/javascript/data-collector.js") %>`

**Comment by lerni** [comment]
No, the module doesn't automatically intercept scripts loaded via `Requirements::javascript()` or `<% require javascript() %>`.

Klaro needs to control when scripts execute based on user consent. Scripts loaded through Requirements are added directly to the page and execute immediately, bypassing consent management.

For consent-managed scripts, special data-attributes are needed:
- `type="text/plain"` (prevents immediate execution)
- `data-type="text/javascript"` (tells Klaro the real type)  
- `data-name="service-name"` (links to your consent config)
- `data-src="script-url"` (the actual script URL)

Auto-converting Requirements-loaded scripts is out of scope ATM. While it's a noted [todo](https://github.com/lerni/klaro-cookie-consent?tab=readme-ov-file#todo) **_add template-parser to add data-attributes_**, it's not currently prioritized, so unlikely to happen :wink:
